keel-artifact/src/main/kotlin/com/netflix/spinnaker/keel/artifacts/ArtifactListener.kt

```diff
@@ -55,12 +59,10 @@ class ArtifactListener(
         if (repository.isRegistered(artifact.name, artifact.artifactType)) {
           val artifactSupplier = artifactSuppliers.supporting(artifact.artifactType)
           if (artifactSupplier.shouldProcessArtifact(artifact)) {
-            val enrichedArtifact = artifactSupplier.addMetadata(artifact.normalized())
-
             log.info("Registering version {} (status={}) of {} artifact {}",
               artifact.version, artifact.status, artifact.type, artifact.name)
 
-            repository.storeArtifactVersion(enrichedArtifact)
+            enrichAndStore(artifact.normalized(), artifactSupplier)
```

Comment:
changed this to a funciton so we correctly do the chain of things (enrich, send event, and store) in the 3 places it's needed